lib/rack/lint.rb

```diff
@@ -758,21 +758,53 @@ def verify_content_length(size)
       ## that yields +String+ instances, a +Proc+ instance, or a File-like
       ## object.
       ##
-      ## The Body must respond to +each+ or +call+. It may optionally respond to
-      ## +to_path+.
+      ## The Body must respond to +each+ or +call+. It may optionally respond
+      ## to +to_path+ or +to_ary+. A Body that responds to +each+ is considered
+      ## to be an Enumerable Body. A Body that responds to +call+ is considered
+      ## to be a Streaming Body.
       ##
-      ## A Body that responds to +each+ is considered to be an Enumerable Body.
-      ##
-      ## A Body that responds to +call+ is considered to be a Streaming Body.
-      ##
-      ## A Body that responds to +to_path+ is expected to generate the same
-      ## content as would be produced by reading a local file opened with the
-      ## path returned by calling +to_path+.
-      ##
-      ## A body that responds to both +each+ and +call+ must be treated as an
+      ## A Body that responds to both +each+ and +call+ must be treated as an
       ## Enumerable Body, not a Streaming Body. If it responds to +each+, you
-      ## must call +each+ and not +call+. If the body doesn't respond to
+      ## must call +each+ and not +call+. If the Body doesn't respond to
       ## +each+, then you can assume it responds to +call+.
+      ##
+      ## The Body must either be consumed or returned. The Body is consumed by
+      ## optionally calling one of +each+, +call+, +to_path+, or +to_ary+.
+      ## Then, if the Body responds to +close+, it must be called to release
```

Comment:
Well, `File` responds to `to_path` and if it's not closed, I assume it leaks?

Reply:
That's a good point.  We should probably specify that `close` should be called in the `to_path` case, but not in the `to_ary` case (since it is called implicitly in that case).

Reply:
Okay, I think I've clarified this.